pkg/apitype/events.go

```diff
@@ -40,9 +40,20 @@ type DiagnosticEvent struct {
 	Message string `json:"message"`
 	Color   string `json:"color"`
 	// Severity is one of "info", "info#err", "warning", or "error".
-	Severity  string `json:"severity"`
-	StreamID  int    `json:"streamID,omitempty"`
-	Ephemeral bool   `json:"ephemeral,omitempty"`
+	Severity        string           `json:"severity"`
+	StreamID        int              `json:"streamID,omitempty"`
+	Ephemeral       bool             `json:"ephemeral,omitempty"`
+	PolicyViolation *PolicyViolation `json:"policyViolation,omitempty"`
```

Comment:
I would strongly consider creating a new event type (PolicyViolationEvent) instead of reusing diagnostic event here, that just contains what's in PolicyViolation today as well as a URN.